src/components/Pagination.tsx

```diff
@@ -0,0 +1,154 @@
+import classnames from 'classnames';
+import { EventHandler, SyntheticEvent, useEffect, useState } from 'react';
```

Comment:
If only the types are being used maybe break them out for clarity:

```suggestion
import type { EventHandler, SyntheticEvent} from 'react';
import { useEffect, useState } from 'react';
```